File: gm-utils/js/src/commom/utils/idText/IdTextCompare.js
import IntegerCompare from '../integer/IntegerCompare';
import Null from '../object/Null';

export default class IdTextCompare {

	static isNull(o) {
		return Null.is(o) || Null.is(o.id);
	}

	static eq(a, b) {

		if (IdTextCompare.isNull(a)) {
			return IdTextCompare.isNull(b);
		}
		if (IdTextCompare.isNull(b)) {
			return false;
		} else {
			return IntegerCompare.eq(a.id, b.id);
		}

	}

	static compare(a, b) {

		if (IdTextCompare.isNull(a)) {
			if (IdTextCompare.isNull(b)) {
				return 0;
			} else {
				return -1;
			}
		}
		if (IdTextCompare.isNull(b)) {
			return 1;
		} else {
			return IntegerCompare.compare(a.id, b.id);
		}

	}

}

File: gm-utils/src/main/java/src/commom/utils/idText/IdTextCompare.java
package src.commom.utils.idText;

import src.commom.utils.integer.IntegerCompare;
import src.commom.utils.object.Null;

public class IdTextCompare {

	public static boolean isNull(IdText o) {
		return Null.is(o) || Null.is(o.getId());
	}

	public static boolean eq(IdText a, IdText b) {

		if (isNull(a)) {
			return isNull(b);
		}
		if (isNull(b)) {
			return false;
		} else {
			return IntegerCompare.eq(a.getId(), b.getId());
		}

	}

	public static int compare(IdText a, IdText b) {

		if (isNull(a)) {
			if (isNull(b)) {
				return 0;
			} else {
				return -1;
			}
		}
		if (isNull(b)) {
			return 1;
		} else {
			return IntegerCompare.compare(a.getId(), b.getId());
		}

	}

}
